pyo3-macros-backend/src/pyfunction.rs

```diff
@@ -27,6 +27,11 @@ pub struct PyFunctionAttr {
     pub pass_module: bool,
 }
 
+#[derive(Clone, PartialEq, Debug)]
+pub struct PyFunctionArgAttr {
+    pub from_py_with: Option<syn::ExprPath>,
+}
```

Comment:
I think just `Option<ExprPath>` is not so bad in readability, though either is fine.